api/src/main/java/jakarta/data/FilterType.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) 2024 Contributors to the Eclipse Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ */
+package jakarta.data;
+
+
+/**
+ * Enum representing common filter types used to compare, match, or restrict values in Jakarta Data queries.
+ * The {@code FilterType} enum defines a set of operators to specify how values should be compared or matched
+ * in data filtering, providing flexibility for various querying needs across different contexts.
+ */
+public enum FilterType {
```

Comment:
I put looking only for NoSQL, but for sure, we could include SQL and put as UnsupportedException when it does not work on NoSQL databases.